lib/seed_migration/migrator.rb

```diff
@@ -252,10 +252,20 @@ def self.create_seed_file
 #
 # It's strongly recommended to check this file into your version control system.
 
+# Ignore exceptions when appending to an HABTM association
+def safe_habtm_append(model_associations, associated_model_instance_arr)
+  begin
+    model_associations << associated_model_instance_arr
+  rescue ActiveRecord::RecordNotUnique => e
+
+  end
+end
```

Comment:
I added this method because in the app I was working in, we have active record callbacks around create that automatically add some HABTM associations.  so as an example I was getting this error because I would try to add habtm associations for `ids = [1,2,3]`, but `[1,2]` were already associated, and I would get a mysql exception.  

Using this method allowed the seeds to run successfully, but there was a problem that some associations never get created.  so in the example above, id 3 never gets associated.  I've solved this in https://github.com/pboling/seed_migration/pull/80/commits/76e723b6a8c03debdf72400b1bfa0bbfb4930d33, so this method is technically unnecessary, but it might be helpful to catch still and log.

Reply:
Agreed.